src/isomorphic/children/ReactChildren.js

```diff
@@ -117,7 +117,7 @@ function mapSingleChildIntoContext(bookKeeping, child, childKey) {
         // traverseAllChildren used to do for objects as children
         keyPrefix +
         (
-          mappedChild !== child ?
+          mappedChild !== child && mappedChild.key ?
           escapeUserProvidedKey(mappedChild.key || '') + '/' :
```

Comment:
Yes, that make sense, need to add a null-check then. Still need a truthy check on `mappedChild.key`, so `mappedChild.key && (!child || mappedChild.key !== child.key)`